doc/source/whatsnew/v0.18.1.txt

```diff
@@ -516,3 +516,4 @@ Bug Fixes
 - Bug in ``.describe()`` resets categorical columns information (:issue:`11558`)
 - Bug where ``loffset`` argument was not applied when calling ``resample().count()`` on a timeseries (:issue:`12725`)
 - ``pd.read_excel()`` now accepts column names associated with keyword argument ``names``(:issue:`12870`)
+- Compat with SciPy > 0.17 for deprecated 'piecewise_polynomial' interpolation method (:issue:`12887`)
```

Comment:
put this in Other API changes section. Add a line in Enhancements for the addition of ``from_derivatives`` method as well (you can use the PR number there).